test/performance/range/view/view_all_benchmark.cpp

```diff
@@ -0,0 +1,82 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+// -----------------------------------------------------------------------------------------------------
+
+#include <deque>
+#include <list>
+#include <string>
+#include <vector>
+
+#include <benchmark/benchmark.h>
+
+#include <seqan3/range/view/view_all.hpp>
+
+using namespace seqan3;
+
+
+// ============================================================================
+//  push_back
+// ============================================================================
+
+template <typename container_t, typename adaptor_t>
+void iterate_left_to_right(benchmark::State& state)
+{
+    container_t c;
+    c.resize(1'000'000);
+    uint8_t i = 0;
+    for (auto & e : c)
+        e = ++i; // dummy values
+
+    unsigned dummy = 0;
+
+    if constexpr (std::Same<adaptor_t, void>)
+    {
+        for (auto _ : state)
+        {
+            for (auto e : c)
+                dummy += e;
+        }
+
+    } else
+    {
+        adaptor_t adaptor;
+        auto v = c | adaptor;
+
+        for (auto _ : state)
+        {
+            for (auto e : v)
+                dummy += e;
+        }
+    }
+
+    [[maybe_unused]] volatile unsigned dummy2 = dummy;
+}
+
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, void);
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(std::ranges::view::all));
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(::ranges::view::all));
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::string, decltype(seqan3::view::all));
+
+BENCHMARK_TEMPLATE(iterate_left_to_right, std::vector<int>, void);
```

Comment:
Also here: `int` ?

Reply:
There is no other integer that you can get by writing three letters :wink:

Reply:
Actually math operations are fastest for `size_t` so I have changed everything to that (so that the math operations ahve the smallest possible effect).